lnd_test.go

```diff
@@ -4466,29 +4470,33 @@ var testsCases = []*testCase{
 func TestLightningNetworkDaemon(t *testing.T) {
 	ht := newHarnessTest(t)
 
+	seenTxns := make(chan *chainhash.Hash, 50)
+
+	// First create an instance of the btcd's rpctest.Harness. This will be
+	// used to fund the wallets of the nodes within the test network and to
+	// drive blockchain related events within the network. Revert the default
+	// setting of accepting non-standard transactions on simnet to reject them.
+	// Transactions on the lightning network should always be standard to get
+	// better guarantees of getting included in to blocks.
+	args := []string{"--rejectnonstd"}
+	handlers := &rpcclient.NotificationHandlers{
+		OnTxAccepted: func(hash *chainhash.Hash, amt btcutil.Amount) {
+			seenTxns <- hash
```

Comment:
Sending over a raw channel here which is then passed into the `lnd` harness seems like a step backwards. We can still re-use the same callback method added as we'll still _start_ the btcd harness before we start the lnd harness.

Reply:
Reworked this.